library/common/main_interface.h

```diff
@@ -83,6 +83,14 @@ envoy_status_t reset_stream(envoy_stream_t stream);
  */
 envoy_engine_t init_engine();
 
+/**
+ * Update the network interface to the preferred network for opening new streams.
```

Comment:
Does setting this also kill existing streams? I would assume so since on iOS they'll hang until they time out in the case of wifi <> cellular

Reply:
We intentionally will not kill streams. Often (but not always) the OS will indicate a new preferred network not because the old one rather became completely unavailable, but rather because some heuristic indicated the other should be favored.

It's more common to have both some level of wifi connectivity and cellular during the switchover (and of course I type "wifi" and "cellular", ah well). Keeping streams open allows requests--that may still be on perfectly healthy connections--to complete.

In the future we may consider re-tuning idle timeouts to be much more aggressive for a non-favored network, but this will require some Envoy changes.